lib/deref.js

```diff
@@ -167,8 +175,15 @@ module.exports = {
         // not throwing an error. We didn't find the reference - generate a dummy value
         return { value: 'reference ' + schema.$ref + ' not found in the OpenAPI spec' };
       }
-      if (schemaResolutionCache[refKey]) {
-        return schemaResolutionCache[refKey];
+
+      /**
+       * use cached schema if it was resolved at level lower or equal then at current stack level or
+       * cached schema is resolved fully (it does not contain ERR_TOO_MANY_LEVELS value in sub schema)
+       */
+      if (_.get(schemaResolutionCache, [refKey, 'schema']) &&
+        (_.get(schemaResolutionCache, [refKey, 'maxStack'], 0) < stackLimit ||
+        _.get(schemaResolutionCache, [refKey, 'resLevel'], stackLimit) <= stack)) {
```

Comment:
Can we break this down a bit? Let's have an initial if statement and then declare consts in the first if block and use those consts to compare in the subsequent if block

```
if (_.get(schemaResolutionCache, [refKey, 'schema']) {
    const maxStack = _.get(schemaResolutionCache, [refKey, 'maxStack'], 0),
        resLevel = _.get(schemaResolutionCache, [refKey, 'resLevel'], stackLimit);

       // some comment
        if (maxStack < stackLimit || resLevel <= stack) {

        }

}

Reply:
Sure, will update it.